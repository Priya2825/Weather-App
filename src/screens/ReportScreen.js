import { Text, View, StyleSheet, FlatList } from "react-native";
import HorizontalFlatlist from "../../components/HorizontalFlatlist";
import NextForecastCard from "../../components/NextForecastCard";
import weatherData from "../../assets/weatherData";

export default function ReportScreen({ route, navigation }) {
  const { data, name } = route.params;

  return (
    <View style={styles.container}>
      {/* Forecast Report Header */}
      <Text style={styles.headerText}>Forecast report</Text>

      {/* Today Section */}
      <View style={styles.sectionContainer}>
        <Text style={styles.sectionTitle}>Today</Text>
        <HorizontalFlatlist
          testID="listitem"
          hourlyForecast={data.slice(0, 7)}
        />
      </View>

      {/* Next Forecast Section */}
      <View style={styles.sectionContainer}>
        <Text style={styles.sectionTitle}>Next forecast: {name}</Text>
        <FlatList
          data={data}
          renderItem={({ item }) => (
            <NextForecastCard
              testID="listitem"
              dt_txt={new Date(item.dt * 1000).toISOString().split("T")[0]}
              temperature={item.main.temp}
              currentWeather={item.weather[0].main}
              // handleOnPress={(color, weatherIcon) => {
              //   navigation.push('Detail', { backgroundColor: color, weatherIcon: weatherIcon, ...item });
              // }}
            />
          )}
          keyExtractor={(item, index) => index.toString()}
          contentContainerStyle={styles.listContainer}
          ItemSeparatorComponent={() => <View style={styles.separator} />}
        />
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 16,
    backgroundColor: "#f7f7f7",
  },
  headerText: {
    fontSize: 24,
    fontWeight: "bold",
    textAlign: "center",
    marginBottom: 16,
  },
  sectionContainer: {
    marginBottom: 16,
  },
  sectionTitle: {
    fontSize: 20,
    fontWeight: "600",
    marginBottom: 8,
  },
  listContainer: {
    paddingVertical: 8,
  },
  separator: {
    height: 16,
  },
});
