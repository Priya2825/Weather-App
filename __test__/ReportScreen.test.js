import React from "react";
import { render, screen } from "@testing-library/react-native";
import ReportScreen from "../src/screens/ReportScreen";
import { NavigationContainer } from "@react-navigation/native";

// Mock Data
const mockData = [
  {
    dt: 1636339200,
    main: { temp: 22 },
    weather: [{ main: "Clear" }],
  },
  {
    dt: 1636425600,
    main: { temp: 24 },
    weather: [{ main: "Cloudy" }],
  },
  // Add more mock items if needed
];

const mockRoute = {
  params: {
    data: mockData,
    name: "Test City",
  },
};

const mockNavigation = {
  navigate: jest.fn(),
};

describe("ReportScreen", () => {
  it("renders correctly with given data", () => {
    const tree = render(
      <NavigationContainer>
        <ReportScreen route={mockRoute} navigation={mockNavigation} />
      </NavigationContainer>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("displays the header text", () => {
    render(
      <NavigationContainer>
        <ReportScreen route={mockRoute} navigation={mockNavigation} />
      </NavigationContainer>
    );

    expect(screen.getByText("Forecast report")).toBeTruthy();
  });

  it("displays the today section with hourly forecast", () => {
    render(
      <NavigationContainer>
        <ReportScreen route={mockRoute} navigation={mockNavigation} />
      </NavigationContainer>
    );

    expect(screen.getByText("Today")).toBeTruthy();
    // Assuming HorizontalFlatlist renders correctly
  });

  it("displays the next forecast section with data", () => {
    render(
      <NavigationContainer>
        <ReportScreen route={mockRoute} navigation={mockNavigation} />
      </NavigationContainer>
    );

    expect(screen.getByText("Next forecast: Test City")).toBeTruthy();
    expect(screen.getByText("2021-11-08")).toBeTruthy(); // Replace with date based on mockData
    expect(screen.getByText("22Â°C")).toBeTruthy(); // Replace with temperature based on mockData
    expect(screen.getByText("Clear")).toBeTruthy(); // Replace with weather based on mockData
  });

  it("renders the flat list correctly", () => {
    render(
      <NavigationContainer>
        <ReportScreen route={mockRoute} navigation={mockNavigation} />
      </NavigationContainer>
    );

    expect(screen.getAllByTestId("listitem").length).toBe(mockData.length);
  });
});
