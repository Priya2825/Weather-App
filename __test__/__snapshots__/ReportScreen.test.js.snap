// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportScreen renders correctly with given data 1`] = `
<View
  style={
    {
      "backgroundColor": "#f7f7f7",
      "flex": 1,
      "padding": 16,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 24,
        "fontWeight": "bold",
        "marginBottom": 16,
        "textAlign": "center",
      }
    }
  >
    Forecast report
  </Text>
  <View
    style={
      {
        "marginBottom": 16,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "600",
          "marginBottom": 8,
        }
      }
    >
      Today
    </Text>
    <RCTScrollView
      data={
        [
          {
            "dt": 1636339200,
            "main": {
              "temp": 22,
            },
            "weather": [
              {
                "main": "Clear",
              },
            ],
          },
          {
            "dt": 1636425600,
            "main": {
              "temp": 24,
            },
            "weather": [
              {
                "main": "Cloudy",
              },
            ],
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={0.0001}
      showsHorizontalScrollIndicator={false}
      stickyHeaderIndices={[]}
      style={
        {
          "width": "100%",
        }
      }
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={
            [
              {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "backgroundColor": "#FBFFF1",
                  "borderRadius": 20,
                  "justifyContent": "center",
                  "marginRight": 15,
                  "padding": 10,
                  "width": 120,
                },
                {
                  "backgroundColor": undefined,
                },
              ]
            }
          >
            <Text
              style={
                {
                  "color": "#3C3744",
                  "fontSize": 18,
                  "marginBottom": 5,
                }
              }
            />
            <Image
              source={
                {
                  "uri": undefined,
                }
              }
              style={
                {
                  "height": 50,
                  "resizeMode": "contain",
                  "width": 50,
                }
              }
            />
            <Text
              style={
                {
                  "color": "#3C3744",
                  "fontSize": 18,
                  "fontWeight": "500",
                  "marginTop": 5,
                }
              }
            >
              °C
            </Text>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={
            [
              {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "backgroundColor": "#FBFFF1",
                  "borderRadius": 20,
                  "justifyContent": "center",
                  "marginRight": 15,
                  "padding": 10,
                  "width": 120,
                },
                {
                  "backgroundColor": undefined,
                },
              ]
            }
          >
            <Text
              style={
                {
                  "color": "#3C3744",
                  "fontSize": 18,
                  "marginBottom": 5,
                }
              }
            />
            <Image
              source={
                {
                  "uri": undefined,
                }
              }
              style={
                {
                  "height": 50,
                  "resizeMode": "contain",
                  "width": 50,
                }
              }
            />
            <Text
              style={
                {
                  "color": "#3C3744",
                  "fontSize": 18,
                  "fontWeight": "500",
                  "marginTop": 5,
                }
              }
            >
              °C
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      {
        "marginBottom": 16,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "600",
          "marginBottom": 8,
        }
      }
    >
      Next forecast: 
      Test City
    </Text>
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      contentContainerStyle={
        {
          "paddingVertical": 8,
        }
      }
      data={
        [
          {
            "dt": 1636339200,
            "main": {
              "temp": 22,
            },
            "weather": [
              {
                "main": "Clear",
              },
            ],
          },
          {
            "dt": 1636425600,
            "main": {
              "temp": 24,
            },
            "weather": [
              {
                "main": "Cloudy",
              },
            ],
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={0.0001}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "borderRadius": 20,
                "elevation": 5,
                "flex": 1,
                "marginBottom": 16,
                "opacity": 1,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 10,
              }
            }
            testID="listitem"
          >
            <View
              style={
                [
                  {
                    "borderRadius": 20,
                    "flexDirection": "row",
                    "padding": 20,
                  },
                  {
                    "backgroundColor": "#FFFFFF",
                  },
                ]
              }
            >
              <Image
                source={null}
                style={
                  {
                    "height": 100,
                    "resizeMode": "contain",
                    "width": 100,
                  }
                }
              />
              <View
                style={
                  {
                    "flex": 1,
                    "justifyContent": "center",
                    "marginLeft": 16,
                  }
                }
              >
                <Text>
                  2021-11-08
                </Text>
                <Text
                  style={
                    {
                      "color": "#666",
                      "fontSize": 18,
                    }
                  }
                >
                  Clear
                </Text>
                <Text
                  style={
                    {
                      "color": "#000",
                      "fontSize": 32,
                      "fontWeight": "bold",
                    }
                  }
                >
                  22
                  °C
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              {
                "height": 16,
              }
            }
          />
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "borderRadius": 20,
                "elevation": 5,
                "flex": 1,
                "marginBottom": 16,
                "opacity": 1,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 10,
              }
            }
            testID="listitem"
          >
            <View
              style={
                [
                  {
                    "borderRadius": 20,
                    "flexDirection": "row",
                    "padding": 20,
                  },
                  {
                    "backgroundColor": "#FFFFFF",
                  },
                ]
              }
            >
              <Image
                source={null}
                style={
                  {
                    "height": 100,
                    "resizeMode": "contain",
                    "width": 100,
                  }
                }
              />
              <View
                style={
                  {
                    "flex": 1,
                    "justifyContent": "center",
                    "marginLeft": 16,
                  }
                }
              >
                <Text>
                  2021-11-09
                </Text>
                <Text
                  style={
                    {
                      "color": "#666",
                      "fontSize": 18,
                    }
                  }
                >
                  Cloudy
                </Text>
                <Text
                  style={
                    {
                      "color": "#000",
                      "fontSize": 32,
                      "fontWeight": "bold",
                    }
                  }
                >
                  24
                  °C
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
